<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>738</width>
    <height>645</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>QOAuth test application</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_10" stretch="0,0">
      <item>
       <layout class="QFormLayout" name="formLayout">
        <item row="0" column="0">
         <widget class="QLabel" name="labelConsumerKey">
          <property name="text">
           <string>Consumer Key</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="lineEditConsumerKey"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="labelConsumerSecret">
          <property name="text">
           <string>Consumer Secret</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="lineEditConsumerSecret"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="labelSignatureMethod">
          <property name="text">
           <string>Signature Method</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="comboBoxSignatureMethod">
          <item>
           <property name="text">
            <string>HMAC-SHA1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>RSA-SHA1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Plaintext (no signature)</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="QLabel" name="label">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Minimum">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="maximumSize">
         <size>
          <width>16777215</width>
          <height>50</height>
         </size>
        </property>
        <property name="text">
         <string>This is a test application for QOAuth.</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox">
      <property name="title">
       <string>Request Token</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <widget class="QLabel" name="labelDescRequestToken">
         <property name="text">
          <string>This is a temporary request token needed to be authorised and later exchanged for an access token</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="labelRequestURLRequestToken">
           <property name="text">
            <string>Request URL</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditRequestURLRequestToken">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <item>
            <widget class="QLabel" name="labelRequestToken">
             <property name="text">
              <string>Request Token</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEditRequestToken">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_3">
           <item>
            <widget class="QLabel" name="labelRequestTokenSecret">
             <property name="text">
              <string>Request Token Secret</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLineEdit" name="lineEditRequestTokenSecret">
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_13" stretch="1,10">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_6">
           <item>
            <widget class="QLabel" name="labelHttpMethodRequestToken">
             <property name="text">
              <string>Http Method</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBoxHttpMethodRequestToken">
             <item>
              <property name="text">
               <string>GET</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>POST</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="pushButtonRequestToken">
           <property name="text">
            <string>Request Token</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBox_2">
      <property name="title">
       <string>Access Token</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="QLabel" name="labelDescAccessToken">
         <property name="text">
          <string>This is an access token that has to be stored inside the application to sign further requests made by User.</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_5">
         <item>
          <widget class="QLabel" name="labelRequestURLAccessToken">
           <property name="text">
            <string>Request URL</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="lineEditRequestURLAccessToken">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Expanding" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_7">
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_2">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_8">
             <item>
              <widget class="QLabel" name="labelRequestToken_2">
               <property name="text">
                <string>Request Token</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEditRequestToken_2">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_9">
             <item>
              <widget class="QLabel" name="labelRequestTokenSecret_2">
               <property name="text">
                <string>Request Token Secret</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEditRequestTokenSecret_2">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
         <item>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_11">
             <item>
              <widget class="QLabel" name="labelAccessToken">
               <property name="text">
                <string>Access Token</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEditAccessToken">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QHBoxLayout" name="horizontalLayout_12">
             <item>
              <widget class="QLabel" name="labelAccessTokenSecret">
               <property name="text">
                <string>Access Token Secret</string>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QLineEdit" name="lineEditAccessTokenSecret">
               <property name="readOnly">
                <bool>true</bool>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_15" stretch="1,10">
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout_14">
           <item>
            <widget class="QLabel" name="labelHttpMethodAccessToken">
             <property name="text">
              <string>Http Method</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QComboBox" name="comboBoxHttpMethodAccessToken">
             <item>
              <property name="text">
               <string>GET</string>
              </property>
             </item>
             <item>
              <property name="text">
               <string>POST</string>
              </property>
             </item>
            </widget>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QPushButton" name="pushButtonAccessToken">
           <property name="text">
            <string>Access Token</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <layout class="QHBoxLayout" name="horizontalLayout_16" stretch="1,1,1,1,1">
      <item>
       <spacer name="horizontalSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonClearFields">
        <property name="text">
         <string>Clear Fields</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonRestoreDefaults">
        <property name="text">
         <string>Restore Defauts</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="pushButtonQuit">
        <property name="text">
         <string>Quit</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="horizontalSpacer_2">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
        <property name="sizeHint" stdset="0">
         <size>
          <width>40</width>
          <height>20</height>
         </size>
        </property>
       </spacer>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>738</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>lineEditRequestTokenSecret</sender>
   <signal>textChanged(QString)</signal>
   <receiver>lineEditRequestTokenSecret_2</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>574</x>
     <y>226</y>
    </hint>
    <hint type="destinationlabel">
     <x>255</x>
     <y>428</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>lineEditRequestToken</sender>
   <signal>textChanged(QString)</signal>
   <receiver>lineEditRequestToken_2</receiver>
   <slot>setText(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>212</x>
     <y>225</y>
    </hint>
    <hint type="destinationlabel">
     <x>226</x>
     <y>393</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
